uikit/views/src/main/java/com/sygic/maps/uikit/views/navigation/speed/SpeedProgressView.kt

```diff
@@ -0,0 +1,216 @@
+        set(value) {
+            field = value
+            segmentForegroundGradient = if (value.size >= 2) createForegroundGradient(value) else null
+            invalidate()
+        }
+
+    @ColorInt
+    var segmentBackgroundColor = getColor(R.color.speedProgressBackgroundColor)
+        set(value) {
+            field = value
+            backgroundPaint.color = value
+            invalidate()
+        }
+
+    var segmentAngle = resources.getInteger(R.integer.speedProgressViewSegmentAngle).toFloat()
+        set(value) {
+            field = value
+            invalidate()
+        }
+
+    var segmentThickness = resources.getDimensionPixelSize(R.dimen.speedProgressViewSegmentThickness)
+        set(value) {
+            field = value
+            foregroundPaint.strokeWidth = value.toFloat()
+            backgroundPaint.strokeWidth = value.toFloat()
+            invalidate()
+        }
+
+    var segmentSpacing = resources.getFraction(R.fraction.speedProgressViewSegmentSpacing, 1, 1)
+        set(value) {
+            field = value
+            invalidate()
+        }
+
+    init {
+        attrs?.let { attributeSet ->
+            @Suppress("Recycle")
+            context.obtainStyledAttributes(
+                attributeSet,
+                R.styleable.SpeedProgressView,
+                defStyleAttr,
+                defStyleRes
+            ).also {
+                roundSegmentEdges = it.getBoolean(
+                    R.styleable.SpeedProgressView_roundSegmentEdges, roundSegmentEdges
+                )
+                segmentBackgroundColor = it.getInt(
+                    R.styleable.SpeedProgressView_segmentBackgroundColor, segmentBackgroundColor
+                )
+                segmentForegroundColors = resources.getIntArray(
+                    it.getResourceId(
+                        R.styleable.SpeedProgressView_segmentForegroundColors, R.array.speedProgressForegroundColors
+                    )
+                )
+                segmentAngle = it.getInt(
+                    R.styleable.SpeedProgressView_segmentAngle, segmentAngle.toInt()
+                ).toFloat()
+                segmentThickness = it.getDimensionPixelSize(
+                    R.styleable.SpeedProgressView_segmentThickness, segmentThickness
+                )
+                segmentSpacing = it.getFraction(
+                    R.styleable.SpeedProgressView_segmentSpacing, 1, 1, segmentSpacing
+                )
+            }.recycle()
+        }
+    }
+
+    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
+        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
+
+        val min = min(
+            getDefaultSize(suggestedMinimumWidth, widthMeasureSpec),
+            getDefaultSize(suggestedMinimumHeight, heightMeasureSpec)
+        )
+        setMeasuredDimension(min, min)
+
+        segmentForegroundGradient = createForegroundGradient(segmentForegroundColors)
```

Comment:
`segmentForegroundColors` hore checkujes na dlzku 2, ale tu nie. To znamena, ze tu ti ten gradient vytvori vzdy. Otazne je, ci potrebujes, aby to pole bolo dlzky 2. Nefunguje aj gradient s 1 farbou (aj ked technicky nie je gradient)?

Reply:
pravda, chybí mi tam ten check, jinak to v konstruktoru toho gradientu hodí IllegalArgumentException:
```java
if (colors.length < 2) {
    throw new IllegalArgumentException("needs >= 2 number of colors");
}
```